syntax = "proto3";


option objc_class_prefix = "OP";

package option_params;

service OptionParamsService {
  rpc GetRecord(Request) returns (OptionParams) {}
  rpc AddRecord(OptionParams) returns(Empty){}
}

message OptionParams {
    string exchange = 1;
    double underlyingConId = 2;
    string tradingClass = 3;
    int32 multiplier = 4;
    repeated string expirations = 5;
    repeated double strikes = 6;
}

message Request {
    double underlyingConId = 1;
}

message Empty {}

message Timestamp {
    string date = 1;
}

message Result {
    string date = 1;
}